### ISSUE TITLE
Ability to set the default extension language

### ISSUE THREAD

**Issue by AdrianoCahete** [shanalikhan/code-settings-sync#987]
I read here that the localization is automated, It's ok, but the problem is that the automated process generates a lot of non-sense texts.  
I'll try to update my language file, but **I really like** to use all my IDE's with everything in English, even with my OS in Portuguese. I know a lot of people who like this.

**To be honest, is better to keep the english text than with the automated translation.**
_Is way harder with that translation that to read the original text in english._

The automated translation has "wise remarks" like:
- "Gist" translated to something like "Essence"
- "Force Download" to "Download's Force"
- "Download Public Gist" to "Download Public Essence"

And a lot of other mistakes.

💡 **Describe the solution you'd like**
Add an option to select your preferred language,

**Comment by shanalikhan** [comment]
Settings Sync language is configured via VS Code localization.

If you change the `locale.json` `locale` value as `en`, it will use English language for Settings Sync. Let me know if it helped.

You can send me a PR improving the values for keys in your language that will improve language for your region.

https://github.com/shanalikhan/code-settings-sync/blob/master/package.nls.pt-br.json

**Comment by AdrianoCahete** [comment]
My VSCode install was already in english, but the `locale` trick works well. [I followed this help](https://code.visualstudio.com/docs/getstarted/locales#_changing-the-display-language).

But still thinks that we need a easily way to change this, from extensions settings.

**Comment by shanalikhan** [comment]
> we need a easily way to change this, from extensions settings.

Extension should always be aligned to VS Code settings, i wont suggest each extension behave own way to change the settings.

I'm keeping this issue open for some time to see other users response.

**Comment by shanalikhan** [comment]
> Extension should always be aligned to VS Code settings

Your OS is non-english but your code was on english language. Settings Sync should show english language by default.

are you using the latest version of Settings Sync ? (v3.4.1)

**Comment by AdrianoCahete** [comment]
> are you using the latest version of Settings Sync ? (v3.4.1)

Yes, I made a install on new work's laptop and choose to start a new profile to sync from zero.

**Comment by AlbertPrograms** [comment]
For some reason, my Code was set to the "en" locale, yet, the extension needed it to be set again and restart Code.

It's a good idea to match the extension language with the editor's. When the localization is an automatic Google translated text, however, there should be an opt-out setting.

I'd be happy to help create a good translation for the "hu" locale, if you're interested.

**Comment by shanalikhan** [comment]
> I'd be happy to help create a good translation for the "hu" locale, if you're interested.

Feel free to send me PR, i'm always looking to add/improve languages. I'm currently only managing "en" locale. Its the community who is improving the other languages :)

**Comment by horvathcsabazoltan** [comment]
Yes, the extension should follow VS Code language. But if there is no translation for that language, the automated translation MUST be optional.
